src/verifier.h

```diff
@@ -60,87 +60,66 @@ form DeserializeForm(integer &d, const uint8_t* bytes, int int_size)
     return f;
 }
 
-std::vector<form> DeserializeProof(const uint8_t* proof_bytes, int proof_len, integer &D)
+std::vector<form> DeserializeProof(const uint8_t* proof_bytes, int proof_len, int int_size, integer &D)
 {
-    int int_size = (D.num_bits() + 16) >> 4;
     std::vector<form> proof;
     for (int i = 0; i < proof_len; i += 2 * int_size)
     {
-        std::vector<uint8_t> tmp_bytes;
-        for (int j = 0; j < 2 * int_size; j++)
-            tmp_bytes.push_back(proof_bytes[i + j]);
-        proof.emplace_back(DeserializeForm(D, tmp_bytes.data(), int_size));
+        proof.emplace_back(DeserializeForm(D, &(proof_bytes[i]), int_size));
     }
     return proof;
 }
 
-bool CheckProofOfTimeNWesolowskiInner(integer &D, form x, const uint8_t* proof_blob,
-                                      int blob_len, int iters, int int_size,
-                                      std::vector<int> iter_list, int recursion)
+bool CheckProofOfTimeNWesolowski(integer D, integer a, integer b, const uint8_t* proof_blob, int proof_blob_len, uint64_t iterations, uint64 disc_size_bits, int depth)
```

Comment:
It looks like you made this function use a loop rather than being recursive, it looks like that means fewer copies of `integer` and fewer heap allocations. All good. It's not trivial to determine that it's equivalent, but I suppose we had some existing tests for this function, that are still passing

Reply:
ya its a straight port of mariano's function. No recursion in chia code hopefully. too many blown stacks.